### PR TITLE
Date picker

### PR DESCRIPTION
**Pull request by dgsmith2**
Update to subclass `PaFormControlDirective` and completed demo documentation

### CODE REVIEW

**Review by julienCarret — APPROVED** [review]

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.spec.ts [hunk 1]
```diff
@@ -259,47 +291,40 @@ describe('DatePickerComponent', () => {
             // === Verify ===
             expect(event.stopPropagation).toHaveBeenCalled();
             expect(event.preventDefault).toHaveBeenCalled();
+            expect(touched).not.toHaveBeenCalled();
         });
 
-        it('should not trap events when popup is not displayed', () => {
+        it('should trap events when input is readonly', () => {
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.spec.ts:297]: It should also prevent the user to manually enter a value in the input when readonly, and apply the proper readonly style, which is currently not the case (I tried it in your branch demo).

I think it's just a matter of propagating the readonly property to `pa-input` in the template.

**Review by mpellerin42 — CHANGES_REQUESTED** [review]
Several things:
- you need to update Pastanaga version in package.json
- you need to add an entry in CHANGELOG
- the readonly property doesn't work as expected
- The month and year dropdown should have a max-height set instead of growing infinitely like they currently do.

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.spec.ts (hunk 1 above)
Comment (dgsmith2) [reply projects/pastanaga-angular/src/lib/date-picker/date-picker.component.spec.ts:297]: You are right.  I botched the merge conflicts when updating my branch with master

**Comment by dgsmith2** [comment]
@mpellerin42 ready for another look!

**Comment by barcafa** [comment]
@mpellerin42 Could you revalidate it works as intended?

**Comment by mpellerin42** [comment]
> @mpellerin42 ready for another look!

So sorry @dgsmith2 and @barcafa ! the long weekend break made me totally forgot about this PR 🤦‍♂️
Happy you were able to merge it anyway at the end!

**Pull request merged**